normandy/recipes/tests/test_models.py

```diff
@@ -579,6 +579,42 @@ def test_disable(self):
         with pytest.raises(EnabledState.NotActionable):
             recipe.latest_revision.disable(user=UserFactory())
 
+    def test_it_publishes_when_enabled(self, settings, mocked_remotesettings):
+        recipe = RecipeFactory(name="Test")
+
+        approval_request = recipe.latest_revision.request_approval(creator=UserFactory())
+        approval_request.approve(approver=UserFactory(), comment="r+")
+        recipe.approved_revision.enable(user=UserFactory())
+
+        mocked_remotesettings.return_value.publish.assert_called_with(recipe)
+
+        # Publishes once when enabled twice.
+        with pytest.raises(EnabledState.NotActionable):
+            recipe.approved_revision.enable(user=UserFactory())
+
+        assert mocked_remotesettings.return_value.publish.call_count == 1
+
+    def test_it_does_not_publish_when_approved(self, settings, mocked_remotesettings):
+        recipe = RecipeFactory(name="Test")
+
+        approval_request = recipe.latest_revision.request_approval(creator=UserFactory())
+        approval_request.approve(approver=UserFactory(), comment="r+")
+
+        assert not mocked_remotesettings.return_value.publish.called
+
+    def test_it_unpublishes_when_disabled(self, settings, mocked_remotesettings):
+        recipe = RecipeFactory(name="Test", approver=UserFactory(), enabler=UserFactory())
+
+        recipe.approved_revision.disable(user=UserFactory())
+
+        mocked_remotesettings.return_value.unpublish.assert_called_with(recipe)
+
+        # Unpublishes once when disabled twice.
+        with pytest.raises(EnabledState.NotActionable):
+            recipe.approved_revision.disable(user=UserFactory())
+
+        assert mocked_remotesettings.return_value.publish.call_count == 1
```

Comment:
Can you add two more tests?

1. A test about when a new revision is approved on a recipe that is already enabled? It should automatically publish the newly approved revision without it needing to be enabled again.
2. A test of a recipe that is enabled, disabled, and then re-enabled?